docs/rules/no-param-reassign.md

```diff
@@ -2,9 +2,11 @@
 
 Assignment to variables declared as function parameters can be misleading and lead to confusing behavior, as modifying function parameters will also mutate the `arguments` object. Often, assignment to function parameters is unintended and indicative of a mistake or programmer error.
 
+This rule can be also configured to fail when function parameters are modified. Side effects on parameters can cause counter-intuitive execution flow and make errors difficult to track down.
+
 ## Rule Details
 
-This rule aims to prevent unintended behavior caused by overwriting function parameters.
+This rule aims to prevent unintended behavior caused by modification of function parameters.
```

Comment:
Personally I like "modify or reassign", but I'm just the kind of guy who
likes to remove all possible ambiguity.
On Mar 29, 2016 8:38 AM, "Isaac Levy" <notifications@github.com> wrote:

> In docs/rules/no-param-reassign.md
> <https://github.com/eslint/eslint/pull/5706#discussion_r57723721>:
>
> >  ## Rule Details
> >
> > -This rule aims to prevent unintended behavior caused by overwriting function parameters.
> > +This rule aims to prevent unintended behavior caused by modification of function parameters.
>
> My hope was that 'modification' covered both cases. wdyt?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/5706/files/cad152ab2e43540a71d84b1b5ecef4e151ef368b#r57723721>
>